### DISCUSSION TITLE
Possible to set default name of a new archive to the current folder's parent folder?

### DISCUSSION THREAD

**Discussion by plymouthvan** [aonez/Keka#1083] [Q&A]
I know there's a setting to set the default name of a new archive containing multiple files to the name of the current folder, but I'm wondering if it's possible to name it after the current folders _parent_ folder. I'm guessing probably not, but if there's a clever way to go about doing this, it would be tremendously helpful!

**Comment by aonez** [comment]
Hi @plymouthvan! Can you add an example?

**Reply by plymouthvan** [reply]
> Hi @plymouthvan! Can you add an example?

Sure, so in my particular case I am often compressing files inside a dated folder, which is inside a folder that is named with a users identifier.

The folder structure looks something like this:

Jane Doe / [Date] / [files to compress]

The archive's file name ultimately needs to be JaneDoe_Date.zip. 

Currently I'm just renaming them by hand, but since the info is in the file path, it would be super helpful if there were a way to grab the parent's parent folder and automatically make them.

**Reply by aonez** [reply]
That's a nice idea to add in the placeholders.
<img width="332" alt="Screenshot 2022-05-11 at 16 10 44" src="https://user-images.githubusercontent.com/480163/167870266-883867d5-e1d2-45dd-85e8-64028462ec16.png">

For example:

- `%p0%` for parent folder, in your example `Date`.
- `%p1%` for parent of the parent folder, in your example `Jane Doe`.
-  ...
- `%pg0%` for the root folder, in your example `Jane Doe`.
- `%pg1%` for the second root folder, in your example `Date`.
- ...
 
Even an all path option:

- `%pf%` i.e. `/Users/Keka/Test/Files` -> `Users-Keka-Test-Files.7z`

**Reply by aonez** [reply]
Just added an issue: https://github.com/aonez/Keka/issues/1101

**Reply by plymouthvan** [reply]
So awesome, thank you for looking into it! ❤️
